compiler/rustc_span/src/lev_distance.rs

```diff
@@ -46,6 +46,49 @@ pub fn lev_distance(a: &str, b: &str, limit: usize) -> Option<usize> {
     (dcol[m] <= limit).then_some(dcol[m])
 }
 
+/// Provides a method name similarity score between two method names. The lower the score,
+/// the closer the match. 0 is an identical match.
+///
+/// Uses the Levenshtein distance between the two strings and attempts to account
+/// for cases where one is a substring of the other.
+///
+/// Returns None if the distance exceeds the limit
+pub fn method_name_similarity_score(a: &str, b: &str, limit: usize) -> Option<usize> {
```

Comment:
We could probably give this a more generic name, `lev_distance_with_substrings` or something like that.

Reply:
Changed to `lev_distance_with_substrings`